dll/win32/shdocvw/CNSCBand.cpp

```diff
@@ -0,0 +1,1682 @@
+
+    CComPtr<IDataObject> pObj;
+    hr = pSrcFolder->GetUIObjectOf(m_hWnd, 1, &pLast, IID_IDataObject, 0, (LPVOID*)&pObj);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return;
+
+    DoDragDrop(pObj, this, dwEffect, &dwEffect);
+}
+
+LRESULT CNSCBand::OnBeginLabelEdit(_In_ LPNMTVDISPINFO dispInfo)
+{
+    // TODO: put this in a function ? (mostly copypasta from CDefView)
+    DWORD dwAttr = SFGAO_CANRENAME;
+    CComPtr<IShellFolder> pParent;
+    LPCITEMIDLIST pChild;
+    HRESULT hr;
+
+    CItemData *info = GetItemData(dispInfo->item.hItem);
+    if (!info)
+        return FALSE;
+
+    hr = SHBindToParent(info->absolutePidl, IID_PPV_ARG(IShellFolder, &pParent), &pChild);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return FALSE;
+
+    hr = pParent->GetAttributesOf(1, &pChild, &dwAttr);
+    if (SUCCEEDED(hr) && (dwAttr & SFGAO_CANRENAME))
+    {
+        m_isEditing = TRUE;
+        m_oldSelected = NULL;
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+HRESULT CNSCBand::_UpdateBrowser(LPCITEMIDLIST pidlGoto)
+{
+    CComPtr<IShellBrowser> pBrowserService;
+    HRESULT hr = IUnknown_QueryService(m_pSite, SID_STopLevelBrowser,
+                                       IID_PPV_ARG(IShellBrowser, &pBrowserService));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    hr = pBrowserService->BrowseObject(pidlGoto, SBSP_SAMEBROWSER | SBSP_ABSOLUTE);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    return S_OK;
+}
+
+LRESULT CNSCBand::OnEndLabelEdit(_In_ LPNMTVDISPINFO dispInfo)
+{
+    CItemData *info = GetItemData(dispInfo->item.hItem);
+    HRESULT hr;
+
+    m_isEditing = FALSE;
+    if (m_oldSelected)
+    {
+        ++m_mtxBlockNavigate;
+        TreeView_SelectItem(m_hwndTreeView, m_oldSelected);
+        --m_mtxBlockNavigate;
+    }
+
+    if (!dispInfo->item.pszText)
+        return FALSE;
+
+    CComPtr<IShellFolder> pParent;
+    LPCITEMIDLIST pidlChild;
+    BOOL RenamedCurrent = _IsCurrentLocation(info->absolutePidl) == S_OK;
+
+    hr = SHBindToParent(info->absolutePidl, IID_PPV_ARG(IShellFolder, &pParent), &pidlChild);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return FALSE;
+
+    CComHeapPtr<ITEMIDLIST> pidlNew;
+    hr = pParent->SetNameOf(m_hWnd, pidlChild, dispInfo->item.pszText, SHGDN_INFOLDER, &pidlNew);
+    if (SUCCEEDED(hr) && pidlNew)
+    {
+        CComPtr<IPersistFolder2> pPersist;
+        hr = pParent->QueryInterface(IID_PPV_ARG(IPersistFolder2, &pPersist));
+        if (FAILED_UNEXPECTEDLY(hr))
+            return FALSE;
+
+        CComHeapPtr<ITEMIDLIST> pidlParent;
+        hr = pPersist->GetCurFolder(&pidlParent);
+        if (FAILED_UNEXPECTEDLY(hr))
+            return FALSE;
+
+        CComHeapPtr<ITEMIDLIST> pidlNewAbs(ILCombine(pidlParent, pidlNew));
+        if (RenamedCurrent)
+        {
+            _UpdateBrowser(pidlNewAbs);
+        }
+        else
+        {
+            // Tell everyone in case SetNameOf forgot, this causes IShellView to update itself when we renamed a child
```

Comment:
```suggestion
            // Tell everyone if SetNameOf forgot, this causes IShellView to update itself when we rename a child
```
just because the line was a bit longer than the surrounding code and most likely exceeded column 80 already (not that I would have counted that actually)